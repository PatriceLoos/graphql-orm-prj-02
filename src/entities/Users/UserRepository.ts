import { EntityRepository, Repository } from 'typeorm';
import Users from '.';
import CreateUserParams from '../../graphql/resolvers/Users/types/inputs/CreateUser';

@EntityRepository(Users)
export default class UserRepository extends Repository<Users> {
  async getUser() {
    return this.find();
  }

  async createUser(params: CreateUserParams):Promise<any> {
    return this.save(params).catch((err)=>{
      console.log(`Error creating user ${
        Object.keys(params)}:${
          Object.values(params)}, with error ${err}`)
    });
  }

  async updateUser(id:string,params: CreateUserParams):Promise<any> {
    return this.update(id,params).catch((err)=>{
      console.log(`Error updating user ${
        Object.keys(params)}:${
          Object.values(params)}, with error ${err}`)
    });
  }

  async deleteUser(param: {}):Promise<any> {
    return this.delete(param).catch((err)=>{
      console.log(`Error deleting user${
        Object.keys(param)}:${
          Object.values(param)}, with error ${err}`)
    });
  }

}
