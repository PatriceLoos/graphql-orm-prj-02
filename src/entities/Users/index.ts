import {
  Entity,
  PrimaryGeneratedColumn,
  Column,
  CreateDateColumn,
} from 'typeorm';

@Entity({ name: 'users' })
export default class Users {
  
  @PrimaryGeneratedColumn()
  id!: number;

  @Column({ type: 'text' })
  name!: string;

  @Column({ type: 'text', nullable: true })
  email!: string;

  @Column({ type: 'date', name: 'birth_date' })
  dob?: string;
  
  @Column({ type: 'text', nullable: true })
  address?: string;

  @Column({ type: 'text' })
  description!: string;  

  @CreateDateColumn({
    type: 'timestamptz',
    name: 'created_at',
    default: () => 'CURRENT_TIMESTAMP',
  })
  createdAt!: Date;

  @CreateDateColumn({
    type: 'timestamptz',
    name: 'updated_at',
    default: () => 'CURRENT_TIMESTAMP',
  })
  updatedAt!: Date;

  @Column({ type: 'text' })
  imageUrl!: string;

}
