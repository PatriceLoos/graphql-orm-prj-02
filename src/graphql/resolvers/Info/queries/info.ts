import { Resolver, Query } from 'type-graphql';
import AppInfo from '../types/objects/Info';
import { environment } from '../../../../environment';

@Resolver()
class InfoResolver {
  @Query(() => AppInfo, { description: 'API info' })
  info(): AppInfo {
    return {
      port: Number(environment.PORT),
      healthy: true,
    };
  }
}

export default InfoResolver;
