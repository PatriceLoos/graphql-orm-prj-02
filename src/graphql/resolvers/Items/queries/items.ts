import ItemsRepository from '../../../../entities/Items/ItemsRepository';
import { Resolver, Query } from 'type-graphql';
import { InjectRepository } from 'typeorm-typedi-extensions';
import Items from '../types/objects/Items';


@Resolver()
export default class GetItemsResolver {
  @InjectRepository(ItemsRepository)
  private itemsRepository!: ItemsRepository;

  @Query(() => [Items], { description: 'List of items' })
  async items(): Promise<Items[]> {
    return this.itemsRepository.getItems();
  }
}
