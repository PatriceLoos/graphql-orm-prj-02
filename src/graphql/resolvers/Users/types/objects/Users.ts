import { ObjectType, Field, Int } from 'type-graphql';

@ObjectType()
export default class Users {
  @Field(() => Int)
  id!: number;

  @Field()
  name!: string;

  @Field({ nullable: true })
  address?: string;

  @Field({ nullable: true })
  email?: string;

  @Field({ nullable: true })
  description?: string;

  @Field({ nullable: true })
  dob?: string;

  @Field({ nullable: true })
  createdAt?: Date;

  @Field({ nullable: true })
  updatedAt?: Date;

  @Field({ nullable: true })
  imageUrl?: string;
}
