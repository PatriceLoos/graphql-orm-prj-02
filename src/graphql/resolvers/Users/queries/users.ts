import { Resolver, Query } from 'type-graphql';
import { InjectRepository } from 'typeorm-typedi-extensions';
import Users from '../types/objects/Users';
import UserRepository from '../../../../entities/Users/UserRepository';

@Resolver()
export default class GetUserResolver {
  @InjectRepository(UserRepository)
  private UserRepository!: UserRepository;

  @Query(() => [Users], { description: 'List of Users' })
  async users(): Promise<Users[]> {
    return this.UserRepository.getUser();
  }
}
