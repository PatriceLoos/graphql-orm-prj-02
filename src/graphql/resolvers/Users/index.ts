import users from './queries/users';
import createUser from './mutations/users';

export default {
  users,
  createUser,
};
