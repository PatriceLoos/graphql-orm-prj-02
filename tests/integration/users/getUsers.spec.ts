import { Connection } from 'typeorm';
import { createTestClient } from 'apollo-server-testing';
import { getUsersQuery } from './__gql__/queries';
import {
  createTestConnection,
  createTestServer,
  clearTestDb,
} from '../../utils';
import Users from '../../../src/entities/Users';

let testConnection: Connection;

beforeAll(async () => {
  testConnection = await createTestConnection();
});

beforeEach(async () => {
  await clearTestDb(testConnection);
});

afterAll(async () => {
  await testConnection.close();
});

describe('Get Users', () => {
  it('should return empty array if no users are registered', async () => {
    const apolloServer = await createTestServer();
    const { query } = createTestClient(apolloServer);

    const res = await query({
      query: getUsersQuery,
    });
    expect(res).toMatchSnapshot('Empty users list');
  });

  it('should return array with one registered user', async () => {
    const user = {
      dob: "2021-05-07",
      name: "John Walker",
      id: 1,
      email: "Johnwalker@email.com",
      address: "934 38 st w, Wolf city, DF, 09080",
      description: "user",
      imageUrl: "http://image.com",
    };

    await testConnection.getRepository(Users).save(user);

    const apolloServer = await createTestServer();
    const { query } = createTestClient(apolloServer);

    const res = await query({
      query: getUsersQuery,
    });
    expect(res).toMatchSnapshot('Filled users list');
  });
});
