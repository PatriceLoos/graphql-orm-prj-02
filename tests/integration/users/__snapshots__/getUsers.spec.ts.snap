// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Get Users should return array with one registered user: Filled users list 1`] = `
Object {
  "data": Object {
    "users": Array [
      Object {
        "address": "934 38 st w, Wolf city, DF, 09080",
        "description": "user",
        "dob": "2021-05-07",
        "email": "Johnwalker@email.com",
        "id": 1,
        "imageUrl": "http://image.com",
        "name": "John Walker",
      },
    ],
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Get Users should return empty array if no users are registered: Empty users list 1`] = `
Object {
  "data": Object {
    "users": Array [],
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
